### ISSUE TITLE
Cron reached maximum iterations

### ISSUE THREAD

**Issue by lucasbraum** [kelektiv/node-cron#386]
On this weekend my application started to crash for no reason on AWS. Maybe it is related to the beginning of summer time in Brazil. Because there wasn't any code changes.

I am getting this error:

```Error while executing callback from define API:  Error: Something went wrong. cron reached maximum iterations.
Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
Time Zone: "" - Cron String: 0 0 0 * * * - UTC offset: -03:00 - current Date: Sat Nov 03 2018 08:00:16 GMT+0000
  at CronTime._getNextDateFrom (/srv/api/node_modules/cron/lib/cron.js:243:12)
  at CronTime.sendAt (/srv/api/node_modules/cron/lib/cron.js:151:17)
  at CronTime.getTimeout (/srv/api/node_modules/cron/lib/cron.js:170:29)
  at CronJob.start (/srv/api/node_modules/cron/lib/cron.js:561:31)
  at new CronJob (/srv/api/node_modules/cron/lib/cron.js:524:10)
  at createJob (/srv/api/src/app/jobs/AutoArchiveDoneTasksJob.js:57:14)
  at Function.define (/srv/api/src/app/jobs/AutoArchiveDoneTasksJob.js:65:21)
  at /srv/api/node_modules/node-injectjs/src/core/define.js:44:24
  at process._tickDomainCallback (internal/process/next_tick.js:129:7)
```  

And I am using the lib this way:

```
module.exports = function (fnResolve) {
  "use strict";

  define([
    '$cron'
  ], ({ CronJob }) => {

    function onTick() {
      // do the stuff I need
    }

    function createJob(hour, onTick) {
      return new CronJob({
        cronTime: `00 00 ${hour} * * *`,
        onTick,
        start: true,
        timeZone: 'America/Sao_Paulo'
      });
    }

    const MainJob = createJob('00', onTick);

    const AuxJob = createJob('06', onTick); // just in case the server was not able to run at 0h

    fnResolve({
      MainJob,
      AuxJob
    });
  });
};
```

**Comment by ncb000gt** [comment]
@lucasbraum what version of the module are you using? Sorry it isn't working as expected.

**Comment by jodevsa** [comment]
@lucasbraum confirmed

**Comment by lucasbraum** [comment]
Sorry forgot to mention.
It is `cron@1.4.1`. On my package.json it is `^1.3.0`.

**Comment by jodevsa** [comment]
Hi @lucasbraum ,

Thank you for reporting this issue. 
You are right, this is related to  DST changes for `Sao_Paulo` and have been previously reported in #381 and thought to be fixed.
This is related to an unexpected behavior of a dependency used for timezone handling `moment-timezone`  which we haven't figured out all cases were this might occur. I've pushed a fix to another branch [DST-handeling-revamp](https://github.com/kelektiv/node-cron/tree/DST-handeling-revamp) that you can try out (will push to master after more tests related to DST are added)

**Comment by lucasbraum** [comment]
Could fixed it by upgrading to version `1.5.0`.

Thanks for the support.
